packages/web/pages/guild/[guildname].tsx

```diff
@@ -5,24 +5,49 @@ import { GuildLinks } from 'components/Guild/GuildLinks';
 import { GuildPlayers } from 'components/Guild/Section/GuildPlayers';
 import { ProfileSection } from 'components/Profile/ProfileSection';
 import { HeadComponent } from 'components/Seo';
-import { QuestFragment, QuestStatus_Enum } from 'graphql/autogen/types';
+import {
+  QuestFragment,
+  QuestStatus_Enum,
+  useGetAdministeredGuildsQuery,
+} from 'graphql/autogen/types';
 import { getQuests } from 'graphql/getQuests';
 import { getGuild, getGuildnames } from 'graphql/queries/guild';
+import { useUser } from 'lib/hooks';
 import {
   GetStaticPaths,
   GetStaticPropsContext,
   InferGetStaticPropsType,
 } from 'next';
 import { useRouter } from 'next/router';
 import Page404 from 'pages/404';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { BoxType } from 'utils/boxTypes';
 import { getGuildCoverImageFull } from 'utils/playerHelpers';
 
 type Props = InferGetStaticPropsType<typeof getStaticProps>;
 
 const GuildPage: React.FC<Props> = ({ guild }) => {
   const router = useRouter();
+  const { user, fetching } = useUser();
+  const [getGuildsResult, getGuilds] = useGetAdministeredGuildsQuery();
+  const [canEdit, setCanEdit] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!fetching && user) {
+      getGuilds({ id: user.id });
+    }
+  }, [fetching, user, getGuilds]);
+
+  useEffect(() => {
+    if (
+      user?.id ===
+      getGuildsResult?.data?.guild_metadata.some(
+        (gm) => gm.guildId === guild?.id,
+      )
+    ) {
+      setCanEdit(true);
+    }
+  }, [user, getGuildsResult, guild]);
```

Comment:
perhaps we can repace this useEffect and `canEdit` useState with a single useMemo hook.

Reply:
good call